### PR TITLE
fix(scheduling): one timezone contract from picker to executed instant (closes #546)

### LINKED ISSUE #546: fix(scheduling): audit & fix timezone discrepancies between displayed times and actual schedules
## What
Audit and fix **timezone discrepancies between what's displayed (SPA) and what's actually scheduled/executed** (DB + Celery). Post scheduled_times are stored/assumed UTC (e.g. `run_scheduler` does `scheduled_time.replace(tzinfo=timezone.utc)` on naive DB values), while the SPA renders times to the user — if the user's timezone isn't applied consistently, the displayed post time ≠ the actual post time ≠ the pre-post engagement window.

## Why it matters
This isn't cosmetic: the "post at golden hour" scheduling AND the pre-post feed-commenting window (`scheduled_time − 15 min`, issue linked below) both key off `scheduled_time`. If display and storage disagree by the user's UTC offset, posts and engagement fire hours off from what the user set — defeating the golden-hour/engagement-hacking intent.

## Scope (audit end-to-end, one consistent TZ contract)
- **Storage:** confirm every `scheduled_time` (posts, newsletter_editions, scheduled_dms) is stored in a single canonical zone (UTC) and read back with explicit tzinfo (no naive-assumed-UTC drift).
- **Scheduling:** `get_post_time`/`get_best_posting_time` and the golden/peak-hour logic — verify they compute in the USER's timezone (Login Location / `get_user_timezone`) then convert to UTC for storage.
- **Display:** the SPA (Content Studio, Review/Schedule, newsletter) renders stored UTC → the user's timezone consistently; the "5-day buffer"/"schedule" views show the same instant the backend will act on.
- **Execution:** `auto_check_scheduled_posts`, newsletter publish, DM scheduler, and pre-post commenting all act on the same canonical instant.
- Add a regression test that a time set in the UI round-trips (UI tz → stored UTC → displayed tz → executed instant) with zero drift across a few offsets/DST.

## Acceptance
- Documented single TZ contract; any naive/assumed-UTC conversions made explicit and user-tz-aware; SPA displays match executed instants; round-trip tests. ≥90% patch.

### PR DESCRIPTION
**Pull request by gitchrisqueen**
## What & why

Audits the display → storage → execution path for scheduled posts, newsletter editions and scheduled DMs, writes the contract down, and fixes the places that broke it.

This wasn't cosmetic. Feed commenting fires at `scheduled_time − 15 min` and profile-viewer DMs at `− 10 min`, so a post whose stored instant disagreed with its displayed time dragged the entire golden-hour engagement window along with it.

**The contract** (`docs/timezone-contract.md`): UTC is the only storage and transport zone; the user's IANA timezone is the only display and input zone; conversion happens exactly twice, at the SPA boundary.

## What was actually broken

**Storage — `insert_scheduled_dm` / `update_scheduled_dm` had no normalization at all.** `mysql-connector` serializes a `datetime` from its wall-clock fields and silently drops `tzinfo`, so an aware `14:00-04:00` landed in the column as `14:00` — four hours early, no error raised. Now every scheduling write goes through the new `db.to_naive_utc()`, which also replaces four copies of the same inline conversion in the post writers.

**Scheduling — a double shift.** `recommend_post_times` bucketed posts by their raw *UTC* hour, but `get_post_time` hands that hour straight back to the scheduler as a *local* time to be converted to UTC for storage. A New York user's 4pm winner (21:00 UTC) came back as "21:00 local" and was stored as 02:00 UTC the next day — wrong hour *and* wrong weekday. `recommend_post_times` now takes `tz=` and buckets in the user's zone; `get_post_time` and `/user/post-stats` pass it.

**Transport — missing `Z`.** Newsletter `scheduled_for` / `next_publish` and the scheduled-DM list were serialized as naive ISO, which a browser parses as *local* time. Both now go through `_utc_iso`.

**Display & input — `datetime-local` in the wrong zone.** The Content Studio edit modal sliced the raw UTC ISO into the picker, so the editor showed a different time than the `formatInTimezone` label directly above it. Every save path used `new Date(v).toISOString()`, which reads the value in the *browser's* zone rather than the user's configured one — wrong whenever Login Location and device disagree (travel, VPN, or deliberately targeting another audience). New `toZonedInputValue` / `zonedInputToUtcIso` helpers (DST-correct via `Intl.formatToParts`) now back Content Studio, ComposePost, ScheduledDMs and NewsletterQueue, each labelled with the zone it's editing in.

Verified as already correct and left alone: the `time_zone='+00:00'` session pin (so `NOW()`/`CURDATE()` are UTC), the content-plan local→UTC conversion, `next_publish_datetime`'s DST-safe localize, and the executor re-attaching UTC for the Celery `eta`.

## Testing

`tests/unit/utilities/test_timezone_contract.py` round-trips a picked wall clock through **UI tz → stored UTC → served ISO → displayed tz → executed instant** across 5 zones × 7 dates — both DST transitions, the ambiguous hour after fall-back, and a half-hour offset — asserting zero drift. It also asserts the pre-post engagement window still lands 15/10 minutes before the *displayed* time, that each write normalizes, and that bucketing is tz-aware.

- 5229 unit tests pass (88 new)
- `tsc -b` clean
- eslint: no new findings (the 6 reported on `ContentStudio.tsx` / `NewsletterQueue.tsx` are identical on `main`)

Closes #546

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/cqc_lem/ui/src/pages/review/ScheduledDMs.tsx [hunk 1]
```diff
@@ -69,7 +69,7 @@ export default function ScheduledDMs({ userTimezone }: { userTimezone: string })
         recipient_profile_url: url.trim(),
         recipient_name: name.trim() || null,
         message: body,
-        scheduled_datetime: new Date(when).toISOString(),
+        scheduled_datetime: zonedInputToUtcIso(when, userTimezone),
         status,
       }),
```
[the remark this answers is not in the document]
Comment (gitchrisqueen) [reply src/cqc_lem/ui/src/pages/review/ScheduledDMs.tsx:74]: Fixed in b0480f9. `createMutation` now takes an already-converted `scheduledUtc: string`, and a new `submit()` helper does the conversion first — on null it flashes "Scheduled date/time is not valid." and never calls the API. This matches the guard `ComposePost.handleSubmit` and the ContentStudio bulk-date action already use.

**Comment by gitchrisqueen** [comment]
✅ CI green, Copilot review addressed & resolved — merging (fast path).
